### DISCUSSION TITLE
Yet another accomplishment: brick-list-search

### DISCUSSION THREAD

**Discussion by amano-kenji** [jtdaugherty/brick#443] [Show and tell]
![demo-01](https://user-images.githubusercontent.com/106365348/220034393-ee512079-24af-4259-b9a3-0f753f0baa69.png) ![demo-02](https://user-images.githubusercontent.com/106365348/220034409-1208e8e2-5429-44b8-bb4b-fd9734cb5a2e.png)

I just published https://hackage.haskell.org/package/brick-list-search which allows anyone to easily search forward or backward for a specific kind of element.

For example, you can skip list separators when you move up or down by searching for a list item that is not a separator.

Do you like it? Do you see yourself using it?

**Comment by jtdaugherty** [comment]
Congrats on a second release! I looked at the code and I noticed that it doesn't use `listFindBy` which might make some of your functions even more straightforward to implement. Did you consider using that function from the List module?

**Reply by amano-kenji** [reply]
I forgot about `listFindBy`. I just read `listFindBy`.
* It uses `toList`. I don't know how fast or slow it is.
* It adds 1 to the current index. In brick-list-search, adding 1 to the current index depends on whether the search includes the current index.
* brick-list-search either searches forward or backward for the first element that passes a test. `listFindBy` only searches forward in two partitions created by splitting at the current index.

**Reply by amano-kenji** [reply]
For searching forward or backward, I created `Searchable` type class.

**Comment by jtdaugherty** [comment]
Okay, cool. Thanks for the context.

**Comment by amano-kenji** [comment]
By the way, is there a way to scroll to the bottom or the top even if the current item is not at the bottom or the top?

If I scroll down and up, the first separator at the top is not shown.

**Comment by jtdaugherty** [comment]
> By the way, is there a way to scroll to the bottom or the top even if the current item is not at the bottom or the top?

Yes; to do that you'll need to scroll the list's viewport in your event handler. You'll need to use [vScrollToEnd](https://hackage.haskell.org/package/brick-1.6/docs/Brick-Main.html#v:vScrollToEnd).

**Reply by amano-kenji** [reply]
Okay.
